src/rank_llm/retrieve_and_rerank.py

```diff
@@ -127,4 +128,36 @@ def retrieve_and_rerank(
             for r in retrieved_results:
                 r.ranking_exec_summary = None
 
+    if num_batches > 1 and isinstance(dataset, list):
+        batch_size = len(dataset) // num_batches
+        batch_results = []
+        for i in range(0, len(dataset), batch_size):
+            batch = dataset[i : i + batch_size]
+            # process batch
+            print(f"Processing batch: {i // batch_size + 1} of {num_batches}:")
+            batch_retrieved_results = Retriever.from_inline_documents(
+                query=query, documents=batch
+            )
+            batch_rerank_results = reranker.rerank(
+                batch_retrieved_results,
+                rank_end=top_k_candidates,
+                window_size=min(window_size, len(batch)),
+                shuffle_candidates=shuffle_candidates,
+                logging=print_prompts_responses,
+                step=step_size,
+            )
+            batch_results.extend(batch_rerank_results)
+        rerank_results = batch_results

```

Comment:
I don't think we are calling `sliding_windows_batched`?  Also, is this batching query?